src/core/base.jl

```diff
@@ -1,85 +1,50 @@
-    try
-        solve_time = getsolvetime(gm.model)
-    catch
-        @warn "there was an issue with getsolvetime() on the solver, falling back on @timed.  This is not a rigorous timing value."
-    end
-
-    return status, solve_time
-end
-
-""
-function run_generic_model(power_file, gas_file, power_model_constructor, gas_model_constructor, solver, post_method; solution_builder = get_solution, kwargs...)
-    power_data    = PowerModels.parse_file(power_file)
-    gas_data      = GasModels.parse_file(gas_file)
-    return run_generic_model(power_data, gas_data, power_model_constructor, gas_model_constructor, solver, post_method; solution_builder = solution_builder, kwargs...)
-end
-
-" Run the optimization on a dictionarized model"
-function run_generic_model(power_data::Dict{String,Any}, gas_data::Dict{String,Any}, power_model_constructor, gas_model_constructor, solver, post_method; power_ref_extensions=[], solution_builder = get_solution, kwargs...)
-    pm, gm = build_generic_model(power_data, gas_data, power_model_constructor, gas_model_constructor, post_method; power_ref_extensions=power_ref_extensions, kwargs...)
-    solution = solve_generic_model(pm, gm, solver; solution_builder = solution_builder)
-    return solution
-end
-
 ""
-function build_generic_model(pfile::String, gfile::String, power_model_constructor, gas_model_constructor, post_method; kwargs...)
-    gas_data = GasModels.parse_file(gfile)
-    power_data = PowerModels.parse_file(pfile)
-    return build_generic_model(power_data, gas_data, power_model_constructor, gas_model_constructor, post_method; kwargs...)
-end
-
-" Dummy post method to do nothing for the individual system"
-function empty_post_method(m; kwargs...)
+function instantiate_model(gfile::String, pfile::String, gtype::Type, ptype::Type, build_method; gext=[], pext=[], kwargs...)
+    gdata, pdata = [_GM.parse_file(gfile), _PM.parse_file(pfile)]
+    return instantiate_model(gdata, pdata, gtype, ptype, build_method; gext=gext, pext=pext, kwargs...)
 end
 
 ""
-function build_generic_model(pdata::Dict{String,Any}, gdata::Dict{String,Any}, power_model_constructor, gas_model_constructor, post_method; power_ref_extensions=[], multinetwork=false, multiconductor=false, kwargs...)
-    gm = GasModels.build_generic_model(gdata, gas_model_constructor, empty_post_method; multinetwork=multinetwork, kwargs...)
-    pm = PowerModels.build_model(pdata, power_model_constructor, empty_post_method; ref_extensions=power_ref_extensions, multinetwork=multinetwork, multiconductor=multiconductor)
+function instantiate_model(gdata::Dict{String,<:Any}, pdata::Dict{String,<:Any}, gtype::Type, ptype::Type, build_method; gext=[], pext=[], kwargs...)
+    # Instantiate the separate (empty) infrastructure models.
+    gm = _GM.instantiate_model(gdata, gtype, m->nothing; ref_extensions=gext)
+    pm = _PM.instantiate_model(pdata, ptype, m->nothing; ref_extensions=pext)
```

Comment:
try passing `gm.model` into this call to avoid the `pm.model = gm.model` call later.